test/unit-tests/function/algebra/resolve.test.js

```diff
@@ -5,6 +5,28 @@ import math from '../../../../src/defaultInstance.js'
 
 import { simplifyAndCompare } from './simplify.test.js'
 
+function emptySources (...args) {
+  args.forEach((item) => {
+    if (item.traverse != null) {
+      emptySourcesFromTree(item)
+    } else if (item.forEach != null) {
+      emptySourcesFromArray(item)
+    }
+  })
+
+  return args
+}
+
+function emptySourcesFromArray (array) {
+  array.forEach((item) => emptySources(item))
+}
+
+function emptySourcesFromTree (tree) {
+  tree.traverse((node) => {
+    node.sources = []
```

Comment:
How about using the immutable methods `.clone({ sources: [] })` and `.transform()` in these helper functions to try out whether the immutable API is workable? (eat your own dog foo 😉 )